agent/background_test.go

```diff
@@ -118,6 +120,24 @@ func (s *BackgroundSuite) TestHeartbeatSometimesFailsDoesNotFailTask() {
 	s.True(end.Sub(start) >= time.Second) // canceled by context
 }
 
+func (s *BackgroundSuite) TestHeartbeatFailsOnTaskConflict() {
+	s.mockCommunicator.HeartbeatShouldConflict = true
+	s.a.opts.HeartbeatInterval = time.Millisecond
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	heartbeat := make(chan string)
+	start := time.Now()
+	go s.a.startHeartbeat(ctx, cancel, s.tc, heartbeat)
+	beat := <-heartbeat
+	end := time.Now()
+	sender := s.sender.GetMessage()
+	s.Equal("Heartbeat received signal to abort task", sender.Message.String())
```

Comment:
I don't feel that strongly about this, but still worth saying - I don't think we need to check for exact log message strings. Checking for exact error message strings and messages isn't wrong, but I avoid doing it unless the message has some really important significance. The downside of unit testing exact log message strings is that it becomes more difficult to change source code log strings without a string matching test failing.